app/services/puppet_fact_parser.rb

```diff
@@ -145,6 +145,22 @@ def boot_timestamp
     uptime_seconds.nil? ? nil : (Time.zone.now.to_i - uptime_seconds.to_i)
   end
 
+  def virtual
```

Comment:
Do we want to reuse this:
https://github.com/theforeman/foreman/blob/f96e9788fde478f4a9c656591c8ac51cd9d00385/app/services/puppet_fact_parser.rb#L91
Note that there it's actually outdated: in the past facts were always strings but nowadays it can actually be proper booleans.